docs/developers/guides/token.md

```diff
@@ -0,0 +1,1287 @@
+ [Success]: ✔ should not mint tokens if new total supply overflows max supply RTrace: +338
+[Event] koinos.contracts.token.mint_event / [ '1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqG' ] / ChkAiCtotTmdMTyQZ/OEFgfmKMEtVCA8jEc3EHs=
+[Event] koinos.contracts.token.transfer_event / [
+  '1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqK',
+  '1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqG'
+] / ChkAiCtotTmdMTyQZ/OEFgfmKMEtVCA8jEc3EhkAiCtotTmdMTyQZ/OEFgfmKMEtVCA8jEc6GAo=
+ [Success]: ✔ should transfer tokens RTrace: +391
+[Event] koinos.contracts.token.mint_event / [ '1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqG' ] / ChkAiCtotTmdMTyQZ/OEFgfmKMEtVCA8jEc3EHs=
+[Contract Exit] 'from' has not authorized transfer
+ [Success]: ✔ should not transfer tokens without the proper authorizations RTrace: +295
+[Event] koinos.contracts.token.mint_event / [ '1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqG' ] / ChkAiCtotTmdMTyQZ/OEFgfmKMEtVCA8jEc3EHs=
+[Contract Exit] Cannot transfer to self
+ [Success]: ✔ should not transfer tokens to self RTrace: +234
+[Event] koinos.contracts.token.mint_event / [ '1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqG' ] / ChkAiCtotTmdMTyQZ/OEFgfmKMEtVCA8jEc3EHs=
+[Contract Exit] 'from' has insufficient balance
+ [Success]: ✔ should not transfer if insufficient balance RTrace: +307
+
+    [File]: assembly/__tests__/Token.spec.ts
+  [Groups]: 2 pass, 2 total
+  [Result]: ✔ PASS
+[Snapshot]: 0 total, 0 added, 0 removed, 0 different
+ [Summary]: 13 pass,  0 fail, 13 total
+    [Time]: 139.55ms
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  [Result]: ✔ PASS
+   [Files]: 1 total
+  [Groups]: 2 count, 2 pass
+   [Tests]: 13 pass, 0 fail, 13 total
+    [Time]: 5130.905ms
+┌───────────────────┬───────┬───────┬──────┬──────┬───────────┐
+│ File              │ Total │ Block │ Func │ Expr │ Uncovered │
+├───────────────────┼───────┼───────┼──────┼──────┼───────────┤
+│ assembly/Token.ts │ 100%  │ 100%  │ 100% │ 100% │           │
+├───────────────────┼───────┼───────┼──────┼──────┼───────────┤
+│ total             │ 100%  │ 100%  │ 100% │ 100% │           │
+└───────────────────┴───────┴───────┴──────┴──────┴───────────┘
+
+```
+
+---
+## Customizing the token
+Let's customize the specifics of our token project by modifying `assembly/Token.ts`. Define the following:
+
+- `_name`: The name of your token
+
+- `_symbol`: The symbol for your token
+
+- `_decimals`: The decimal places for your token
+
+
+```ts linenums="1" title="assembly/Token.ts"
+- _name: string = "My Token Name";
+- _symbol: string = "MTN";
+- _decimals: u32 = 8;
+```
+
+As an example, we changed the `[token name]` variable to `My Token Name` and the `symbol` variable to `MTN`.
+
+
+After making your changes, update the token name and symbol in the unit test file located at `assembly/__tests__/Token.spec.ts`.
+We will need to change the test for the `name` and `symbol` functions to reflect your token name and symbol.
+
+```ts linenums="25" title="assembly/__tests__/Token.spec.ts"
```

Comment:
Good, this shows the context in the source file. I would make the change here and then highlight the modified line. That way the guide shows a correct version in case they just want to copy/paste.

Reply:
Please highlight the modified lines.